src/main/java/graphql/normalized/NormalizedInputValue.java

```diff
@@ -0,0 +1,61 @@
+package graphql.normalized;
+
+import java.util.Objects;
+
+/**
+ * A value with type information.
+ */
+public class NormalizedInputValue {
+    private final String type;
+    private final Object value;
+
+    public NormalizedInputValue(String type, Object value) {
+        this.type = type;
+        this.value = value;
+    }
+
+    /**
+     * This can be a wrapped type: e.g. [String!]!
+     *
+     * @return
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Depending on the type it returns:
+     * Scalar or Enum: the value of the Scalar.
```

Comment:
So is this based on `graphql.schema.Coercing#parseValue`?

If so, can we add a `{@link …}` here

Reply:
not really ... it can also be `parseLiteral`. This will change soon anyway as we want to represent the value (Literal String probably)